backend/Lingva.BusinessLayer/SubtitlesParser/Classes/Parsers/SsaParser.cs

```diff
@@ -0,0 +1,172 @@
+            if (!ssaStream.CanRead || !ssaStream.CanSeek)
+            {
+                var message = string.Format("Stream must be seekable and readable in a subtitles parser. " +
+                                   "Operation interrupted; isSeekable: {0} - isReadable: {1}",
+                                   ssaStream.CanSeek, ssaStream.CanSeek);
+                throw new ArgumentException(message);
+            }
+
+            // seek the beginning of the stream
+            ssaStream.Position = 0;
+
+            var reader = new StreamReader(ssaStream, encoding, true);
+
+            var line = reader.ReadLine();
+            var lineNumber = 1;
+            // read the line until the [Events] section
+            while (line != null && line != EventLine)
+            {
+                line = reader.ReadLine();
+                lineNumber++;
+            }
+
+            if (line != null)
+            {
+                // we are at the event section
+                var headerLine = reader.ReadLine();
+                if (!string.IsNullOrEmpty(headerLine))
+                {
+                    var columnHeaders = headerLine.Split(Separator).Select(head => head.Trim()).ToList();
+
+                    var startIndexColumn = columnHeaders.IndexOf(StartColumn);
+                    var endIndexColumn = columnHeaders.IndexOf(EndColumn);
+                    var textIndexColumn = columnHeaders.IndexOf(TextColumn);
+
+                    if (startIndexColumn > 0 && endIndexColumn > 0 && textIndexColumn > 0)
+                    {
+                        var items = new List<SubtitleItem>();
+
+                        line = reader.ReadLine();
+                        while (line != null)
+                        {
+                            if(!string.IsNullOrEmpty(line))
+                            {
+                                var columns = line.Split(Separator);
+                                var startText = columns[startIndexColumn];
+                                var endText = columns[endIndexColumn];
+
+
+                                var textLine = string.Join(",", columns.Skip(textIndexColumn));
+
+                                var start = ParseSsaTimecode(startText);
+                                var end = ParseSsaTimecode(endText);
+
+                                // TODO: split text line?
+                                if (start > 0 && end > 0 && !string.IsNullOrEmpty(textLine))
+                                {
+                                    var item = new SubtitleItem()
+                                                            {
+                                                                StartTime = start,
+                                                                EndTime = end,
+                                                                Lines = new List<string>() { textLine }
+                                                            };
```

Comment:
A code formatting issue. Move brackets to begin of line. In order to look like:
var item = new SubtitleItem
{
…. 
}